### PR TITLE
ISPN-5174 Transaction cannot be recommitted after ownership changes

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-5174

### CODE REVIEW

File: core/src/test/java/org/infinispan/tx/OptimisticPartialCommitTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Test that the modifications of a transaction that were not committed on a node because it didn't own all the keys
+ * are still applied after the node becomes an owner for all of them.
+ *
+ * @author Dan Berindei
+ */
+@CleanupAfterMethod
+@Test(groups = "functional", testName = "tx.ParticipantFailsAfterPrepareTest")
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/tx/OptimisticPartialCommitTest.java]: wrong test name

**Comment by pruivo** [comment]
@danberindei looks good except the failure in CI :)

**Comment by danberindei** [comment]
Looking at the failure now...

**Comment by danberindei** [comment]
TxReplay2Test should be fixed now. I also fixed a random failure in OptimisticPartialCommitTest, caused by a state transfer tx triggering the `after_commit_on_1` state.

**Comment by galderz** [comment]
Integrated, thanks @danberindei

**Pull request closed without merging**